assets/js/components/setup/SetupUsingProxyWithSignIn/index.js

```diff
@@ -0,0 +1,285 @@
+		title = sprintf(
+			/* translators: %s: is the site's hostname. (e.g. example.com) */
+			__( 'You revoked access to Site Kit for %s', 'google-site-kit' ),
+			punycode.toUnicode( new URL( siteURL ).hostname )
+		);
+		description = __(
+			'Site Kit will no longer have access to your account. If you’d like to reconnect Site Kit, click "Sign in with Google" below to generate new credentials.',
+			'google-site-kit'
+		);
+	} else if (
+		DISCONNECTED_REASON_CONNECTED_URL_MISMATCH === disconnectedReason
+	) {
+		title = __( 'Reconnect Site Kit', 'google-site-kit' );
+		description = __(
+			'Looks like the URL of your site has changed. In order to continue using Site Kit, you’ll need to reconnect, so that your plugin settings are updated with the new URL.',
+			'google-site-kit'
+		);
+
+		getHelpURL = changedURLHelpLink;
+	} else if ( isSecondAdmin ) {
+		title = __(
+			'Connect your Google account to Site Kit',
+			'google-site-kit'
+		);
+		description = __(
+			'Site Kit has already been configured by another admin of this site. To use Site Kit as well, sign in with your Google account which has access to Google services for this site (e.g. Google Analytics). Once you complete the 3 setup steps, you’ll see stats from all activated Google services.',
+			'google-site-kit'
+		);
+		showLearnMoreLink = true;
+	} else {
+		title = __( 'Set up Site Kit', 'google-site-kit' );
+		description = __(
+			'Get insights on how people find your site, as well as how to improve and monetize your site’s content, directly in your WordPress dashboard',
+			'google-site-kit'
+		);
+	}
+
+	return (
+		<Fragment>
+			<SetupHeader />
+			<div className="googlesitekit-setup">
+				<Grid>
+					<Row>
+						<Cell size={ 12 }>
+							<Layout rounded>
+								<Splash
+									title={ title }
+									description={ description }
+									showLearnMoreLink={ showLearnMoreLink }
+									secondAdminLearnMoreLink={
+										secondAdminLearnMoreLink
+									}
+									getHelpURL={ getHelpURL }
+									disconnectedReason={ disconnectedReason }
+									connectedProxyURL={ connectedProxyURL }
+									homeURL={ homeURL }
+									analyticsModuleActive={
+										analyticsModuleActive
+									}
+									analyticsModuleAvailable={
+										analyticsModuleAvailable
+									}
+								>
+									{ ( { complete, inProgressFeedback } ) => (
+										<SetupActions
+											proxySetupURL={ proxySetupURL }
+											onButtonClick={ onButtonClick }
+											goToSharedDashboard={
+												goToSharedDashboard
+											}
+											isSecondAdmin={ isSecondAdmin }
+											hasMultipleAdmins={
+												hasMultipleAdmins
+											}
+											hasViewableModules={
+												hasViewableModules
+											}
+											isResettable={ isResettable }
+											complete={ complete }
+											inProgressFeedback={
+												inProgressFeedback
+											}
```

Comment:
Same here. Let's move the component-specific logic scoped within the component to avoid prop drilling. Only the common logic and state management should remain in the parent component.